src/ripple/shamap/impl/SHAMapSync.cpp

```diff
@@ -525,16 +514,25 @@ SHAMap::addKnownNode (const SHAMapNodeID& node, Blob const& rawNode,
 
             if (newNode && isInconsistentNode(newNode))
             {
-                return SHAMapAddNode::invalid();
+                state_ = SHAMapState::Invalid;
+                return SHAMapAddNode::useful();
```

Comment:
@scottschurr The convention I sometimes use for naming a function to indicate what conditions its called under is "on..." for example `onUsefulData()`.